### ISSUE TITLE
404 when Unit Testing Dingo API Routes

### ISSUE THREAD

**Issue by Deleugpn** [dingo/api#1045]
I've seen issue #571, #540 and a few others, but I have no clue as to how they correlate to my problem because I'm not protecting my routes with any Middlewares nor I'm using WithoutMiddleware trait.

I have a working endpoint routed like this:

```
$api = app('Dingo\Api\Routing\Router');

$api->version('v2', function ($api) {
    
    $api->get('/', function(){
        return 'hi';
    });    

    $api->get('stores', 'App\Http\Controllers\v2\StoresController@index');
    $api->get('stores/{slug}', 'App\Http\Controllers\v2\StoresController@show');
});
```

with these Environment Variables 

```
API_STANDARDS_TREE=x
API_SUBTYPE=prime
API_DOMAIN=api.local.puphpet
API_VERSION=v2
API_NAME="App Name"
API_STRICT=false
API_DEBUG=true
API_DEFAULT_FORMAT=json
```

And I can easily make a call for it using Postman, AngularJS or even a **wget** will work when running **wget http://api.local.puphpet/prime-laravel/public/stores**

But with my Unit Test, I only get 404.

```
<php>
        <env name="APP_ENV" value="testing"/>
        <env name="DB_CONNECTION" value="sqlite_testing"/>
        <env name="API_STANDARDS_TREE" value="x"/>
        <env name="API_SUBTYPE" value="prime"/>
        <env name="API_DOMAIN" value="api.local.puphpet"/>
        <env name="API_VERSION" value="v2"/>
        <env name="API_DEFAULT_FORMAT" value="json"/>
        <env name="API_STRICT" value="false"/>
        <env name="CACHE_DRIVER" value="array"/>
        <env name="SESSION_DRIVER" value="array"/>
        <env name="QUEUE_DRIVER" value="sync"/>
    </php>
```

The function looks like this

```
    /**
     * @tests
     */
    public function it_should_get_all_stores() {
        $collection = factory(Store::class)->times(3)->make();

//        $result = $this->call('GET', 'http://api.local.puphpet/prime-laravel/public/stores', [], [], [], ['Accept' => 'application/x.prime.v2+json']);
//        dd($result);    // Message 404 Not Found

//        $result = $this->get('http://api.local.puphpet/prime-laravel/public/stores');
//        dd($result);    // Message 404 Not Found

        // $this->get('/', ['Accept' => 'application/x.prime.v2+json'])->dump();    // Gives 404
    }
```

None of these options works.

This is my TestCase

```
class TestCase extends Illuminate\Foundation\Testing\TestCase {
    /**
     * The base URL to use while testing the application.
     *
     * @var string
     */
    protected $baseUrl = 'http://api.local.puphpet/prime-laravel/public';

    /**
     * Creates the application.
     *
     * @return \Illuminate\Foundation\Application
     */
    public function createApplication() {
        putenv('DB_CONNECTION=sqlite_testing');

        $app = require __DIR__ . '/../bootstrap/app.php';

        $app->make(Illuminate\Contracts\Console\Kernel::class)->bootstrap();

        return $app;
    }

    public function setUp() {
        parent::setUp();
        Artisan::call('migrate');
    }

    public function tearDown() {
        Artisan::call('migrate:reset');
        parent::tearDown();
    }

}
```

**Comment by Deleugpn** [comment]
After much digging into the `vendor` classes and a lot of die and dump, I realized that the HostValidator was rejecting the calls, so here are my findings:

I run this on a Puphpet / Vagrant / Virtual Box setup with a Linux (CentOS) environment. My host machine is Windows. When I was making calls to the API, I was using Windows, so my hosts file would redirect to the Puphpet machine and it would get routed just fine. When trying to do it using Unit Test, I didn't realize that I had to use 127.0.0.1 to make it work.

I changed my phpunit.xml to 

```
       <env name="API_STANDARDS_TREE" value="x"/>
        <env name="API_SUBTYPE" value="prime"/>
        <env name="API_DOMAIN" value="127.0.0.1"/>
        <env name="API_VERSION" value="v2"/>
        <env name="API_DEFAULT_FORMAT" value="json"/>
```
and in my TestCase I changed the BaseURL like

```
protected $baseUrl = 'http://127.0.0.1';
```

and my call works like this

```
$this->get('/stores', ['Accept' => 'application/x.prime.v2+json'])->dump();
```
